rusk-recovery/src/state.rs

```diff
@@ -243,10 +243,14 @@ fn generate_empty_state<P: AsRef<Path>>(
 }
 
 // note: deploy consumes session as it produces commit id
-pub fn deploy<P: AsRef<Path>>(
+pub fn deploy<P: AsRef<Path>, F>(
     state_dir: P,
     snapshot: &Snapshot,
-) -> Result<(VM, [u8; 32]), Box<dyn Error>> {
+    f: F,
```

Comment:
What is this?

Reply:
gives a chance to deploy caller to perform additional operations on the session, since deploy consumes the session, pls. see the declaration below:
```
where
    F: FnOnce(&mut Session),
```
so it is like a hook which can be used but does not have to